lib/kitchen/directions/bake_chapter_introductions.rb

```diff
@@ -7,9 +7,23 @@ def self.v1(book:, bake_chapter_objectives: true)
         book.chapters.each do |chapter|
           chapter_outline_html = ''
 
-          if bake_chapter_objectives
+          case bake_chapter_objectives
+          when 'div'
+            chapter_objectives = chapter.first('div.chapter-objectives')&.cut
+            if chapter_objectives
+              chapter_objectives.search('div[data-type="title"]')&.trash
+              chapter_objectives.wrap_children('div', class: 'os-note-body')
+              chapter_objectives.prepend(child:
+                <<~HTML
+                  <h3 class="os-title" data-type="title">
+                    <span class="os-title-label">#{I18n.t(:chapter_objectives)}</span>
+                  </h3>
+                HTML
+              )
+              chapter_outline_html = chapter_objectives.cut
+            end
+          when true
```

Comment:
I'm not sure that we can use different types for method parameters :thinking: I think it can't be in one case boolean, and in another one string. I have two ideas:

1. We can use two parameters: `bake_chapter_outline` and `bake_chapter_objectives` of boolean type. Names are based on the final class for the element.
2. We can use `bake_chapter_objectives` parameter but it always be string type. What do you think about using strings: `modify` for the first case, because then chapter objectives exists in assembled file and we're just modifying it a little bit; and for second case it could be `add`, because we are creating new element, that is not exists in assembled file?

Let me know your thoughts :wink: Maybe we can also ask for an opinion second reviewer.

Reply:
Hm, the first seems best, as these deal with two separate things (though I don't see them overlapping). It looks like I overcondensed the code a bit 😅

Reply:
Oh because it would break the past recipes because they both deal with chapter objectives, that's why I went with the case. I think the different types for method parameters work here because of the bottom true/false at the end of the case statements. That end bit can catch whether the parameter exists or not, and won't crash anything. Maybe we should get a second opinion?

Reply:
Oh I see :thinking: If we go with second solution all recipes have to be modified. And it's not what we want to do :smile: I asked @ajluk for an opinion and she said that she had similar problem, and it was resolved the same way as you did. Here is the link to PR: https://github.com/openstax/kitchen/pull/181#discussion_r644975380. There are also cases, and parameter type in method is boolean, but there is a case where it is Symbol.

So I think we can stay with current solution. I will give an approve :wink:

Reply:
I actually talked with JP about this! I'm going to split this off into a v2 that uses symbols instead of strings to set the chapter_objective_strategy, and then depreciate the v1 method!